File: src/test/java/dev/paigewatson/layoutmaster/helpers/CarTypeServiceFake.java
package dev.paigewatson.layoutmaster.helpers;

import dev.paigewatson.layoutmaster.client.services.CarTypeService;
import dev.paigewatson.layoutmaster.models.goods.GoodsType;
import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation;
import dev.paigewatson.layoutmaster.models.rollingstock.CarType;

import java.util.Collections;
import java.util.List;

public class CarTypeServiceFake implements CarTypeService
{
    private List<CarType> returnedCarTypeList;
    private CarType savedEntity;
    private CarType returnedCarTypeWithAAR;
    private List<AARDesignation> returnedAARTypes = Collections.emptyList();

    @Override
    public List<AARDesignation> allAARDesignations()
    {
        return returnedAARTypes;
    }


    @Override
    public CarType saveCarTypeToDatabase(CarType carTypeToSave)
    {
        this.savedEntity = carTypeToSave;
        return carTypeToSave;
    }

    @Override
    public List<CarType> allCarTypes()
    {
        return this.returnedCarTypeList;
    }

    @Override
    public CarType carTypeForAAR(AARDesignation expectedAARDesignation)
    {
        return this.returnedCarTypeWithAAR;
    }

    @Override
    public List<CarType> carTypesThatCarryGoodsType(GoodsType expectedGoodsType)
    {
        return returnedCarTypeList;
    }

    public void setReturnedCarTypeList(List<CarType> returnedCarTypeList)
    {
        this.returnedCarTypeList = returnedCarTypeList;
    }

    public CarType savedDtoEntity()
    {
        return this.savedEntity;
    }

    public void setReturnedCarTypeWithAAR(CarType carType)
    {
        returnedCarTypeWithAAR = carType;
    }

    public void setReturnAARTypes(List<AARDesignation> returnedAARTypes)
    {
        this.returnedAARTypes = returnedAARTypes;
    }
}

File: src/test/kotlin/dev/paigewatson/layoutmaster/helpers/CarTypeServiceFake.kt
package dev.paigewatson.layoutmaster.helpers

import dev.paigewatson.layoutmaster.client.services.CarTypeService
import dev.paigewatson.layoutmaster.models.goods.GoodsType
import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation
import dev.paigewatson.layoutmaster.models.rollingstock.CarType

class CarTypeServiceFake : CarTypeService {
    private var returnedCarTypeList: List<CarType>? = null
    private var savedEntity: CarType? = null
    private var returnedCarTypeWithAAR: CarType? = null
    private var returnedAARTypes: List<AARDesignation> = emptyList()
    override fun allAARDesignations(): List<AARDesignation> {
        return returnedAARTypes
    }

    override fun saveCarTypeToDatabase(carTypeToSave: CarType): CarType {
        savedEntity = carTypeToSave
        return carTypeToSave
    }

    override fun allCarTypes(): List<CarType> {
        return returnedCarTypeList!!
    }

    override fun carTypeForAAR(expectedAARDesignation: AARDesignation): CarType {
        return returnedCarTypeWithAAR!!
    }

    override fun carTypesThatCarryGoodsType(expectedGoodsType: GoodsType): List<CarType> {
        return returnedCarTypeList!!
    }

    fun setReturnedCarTypeList(returnedCarTypeList: List<CarType>?) {
        this.returnedCarTypeList = returnedCarTypeList
    }

    fun savedDtoEntity(): CarType? {
        return savedEntity
    }

    fun setReturnedCarTypeWithAAR(carType: CarType?) {
        returnedCarTypeWithAAR = carType
    }

    fun setReturnAARTypes(returnedAARTypes: List<AARDesignation>) {
        this.returnedAARTypes = returnedAARTypes
    }
}
